api/v1beta1/common.go

```diff
@@ -85,33 +113,83 @@ type NodeSection struct {
 	Nova *NovaTemplate `json:"nova,omitempty"`
 }
 
+// NodeTemplate is a specification of the top level node attributes
+type NodeTemplate struct {
+	// Nodes - Map of Node Names and node specific data. Values here override defaults in the
+	// upper level section.
+	// +kubebuilder:validation:Required
+	Nodes map[string]NodeSection `json:"nodes"`
```

Comment:
I just did it this way from a hierarchy perspective. Spec > NodeTemplate contains all of the common variables that will apply to all nodes > Node contains node specific data that overrides the common variables.

But yeah , it would work either way. Not a lot of effort to change it if we want to after this is merged though.